WPThemeReview/Sniffs/CoreFunctionality/PrefixAllGlobalsSniff.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * WPThemeReview Coding Standard.
+ *
+ * @package WPTRT\WPThemeReview
+ * @link    https://github.com/WPTRT/WPThemeReview
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WPThemeReview\Sniffs\CoreFunctionality;
+
+use WordPressCS\WordPress\Sniffs\NamingConventions\PrefixAllGlobalsSniff as WPCSPrefixAllGlobalsSniff;
+use WordPressCS\WordPress\Sniffs\Files\FileNameSniff;
+
+/**
+ * Extend the PrefixAllGlobalsSniff and silence the errors if coming from template files or folders
+ *
+ * @link  https://github.com/WPTRT/WPThemeReview/issues/205
+ * @link  https://github.com/WPTRT/WPThemeReview/issues/201
+ * @link  https://github.com/WPTRT/WPThemeReview/issues/200
+ *
+ * @since 0.2.0
+ */
+class PrefixAllGlobalsSniff extends WPCSPrefixAllGlobalsSniff {
+
+	/**
+	 * List of allowed folders to check the file path against
+	 *
+	 * @var array
+	 */
+	public $allowed_folders = [
+		'template-parts',
+		'templates',
+		'partials',
+	];
+
+	/**
+	 * Check that defined global variables are prefixed, except if coming from template files.
+	 *
+	 * Overload the process_variable_assignment method from the PrefixAllGlobalsSniff in
+	 * the WordPressCodingStandards ruleset.
+	 *
+	 * @since 0.2.0
+	 *
+	 * @param int $stackPtr The position of the current token in the stack.
+	 *
+	 * @return int|void Integer stack pointer to skip forward or void to continue
+	 *                  normal file processing.
+	 */
+	protected function process_variable_assignment( $stackPtr ) {
+
+		$file = $this->phpcsFile->getFileName();
+		if ( 'STDIN' === $file ) {
+			return;
+		}
+
+		$fileName = basename( $file );
+
+		// Don't process in case of a template file or folder.
+		if ( preg_match( FileNameSniff::THEME_EXCEPTIONS_REGEX, $fileName ) !== 1 && $this->is_from_allowed_folder( $file ) ) {
+			return;
+		} else {
+			parent::process_variable_assignment( $stackPtr );
+		}
+	}
+
+	/**
+	 * Checks if the given file is located in the $allowed_folders array
+	 *
+	 * @param  string $file Full path of the sniffed file.
+	 * @return boolean
+	 */
+	private function is_from_allowed_folder( $file ) {
+		$result = array_filter(
```

Comment:
Thanks for the clarification. I'll replace this with the suggestion. 

Would it change anything if I passed by reference to the file path in the `use` statement? 

I thought it should work. because I'm calling this method on every file, so the `$file` parameter changes every time the sniff is ran. Although I'm probably thinking about the process of sniffing in the wrong way - I'm not redefining the whole thing for each file, just once at the beginning of the sniff process, right? That's why my way didn't work. Still, even if I used pass-by-refference it would still be wrong because of the third point.

Reply:
> Would it change anything if I passed by reference to the file path in the use statement?

Yes, it would, but point 1 and 3 would still stand, so it still wouldn't work.

> I'm not redefining the whole thing for each file, just once at the beginning of the sniff process, right?

Correct.